File: backend/src/main/java/ca/metricalsky/winston/entity/CommentPropertiesEntity.java
package ca.metricalsky.winston.entity;

import jakarta.persistence.Basic;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "comment_properties")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CommentPropertiesEntity {

    @Id
    @Column(name = "comment_id")
    private String commentId;

    @Basic(optional = false)
    @Column(name = "important")
    private boolean important;

    @Basic(optional = false)
    @Column(name = "hidden")
    private boolean hidden;
}

File: database/src/main/kotlin/ca/metricalsky/winston/database/entity/comment/CommentPropertiesEntity.kt
package ca.metricalsky.winston.database.entity.comment

import jakarta.persistence.Basic
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "comment_properties")
class CommentPropertiesEntity(

    @Id
    @Column(name = "comment_id")
    var commentId: String,

    @Basic(optional = false)
    @Column(name = "important")
    var important: Boolean,

    @Basic(optional = false)
    @Column(name = "hidden")
    var hidden: Boolean
)
